src/RoutingApp.php

```diff
@@ -43,9 +40,9 @@ public function call(Context $context) {
 		// Force pages to their designated protocol if specified
 		if ($route->getProtocol() != null) {
 			if ($currentProtocol != $route->getProtocol()) {
-				$url = Url::fromString(sprintf("%s://%s/", $route->getProtocol(), $currentHost))
-						->join($currentUrl)
-						->addParams($_GET);
+				/** @var Url $url */
+				$url = Url::fromString(sprintf("%s://%s%s", $route->getProtocol(), $currentHost, $currentUrl));
```

Comment:
Add Params is removed?

Reply:
This whole thing would have never worked had it actually ever been hit. GET params are empty at this point. Current url has the query params.